rackcity/views/it_instance_views.py

```diff
@@ -173,6 +176,133 @@ def instance_delete(request):
     )
 
 
+@api_view(['POST'])
+@permission_classes([IsAdminUser])
+def instance_bulk_upload(request):
+    """
+    Bulk upload many instances to add or modify
+    """
+    data = JSONParser().parse(request)
+    if 'instances' not in data:
+        return JsonResponse(
+            {"failure_message": "Bulk upload request should have a parameter 'instances'"},
+            status=HTTPStatus.BAD_REQUEST
+        )
+    instance_datas = data['instances']
+    instances_to_add = []
+    potential_modifications = []
+    for instance_data in instance_datas:
+        if (
+            'vendor' not in instance_data
+            or 'model_number' not in instance_data
+        ):
+            return JsonResponse(
+                {"failure_message": "Instance records must include 'vendor' and 'model_number'. "},
+                status=HTTPStatus.BAD_REQUEST
+            )
+        try:
+            model = ITModel.objects.get(
+                vendor=instance_data['vendor'],
+                model_number=instance_data['model_number']
+            )
+        except ObjectDoesNotExist:
+            failure_message = "Model does not exist: " + \
```

Comment:
Should probably include entire row for what fails in these error messages. But not blocking, we can perfect all these later